build.sbt

```diff
@@ -36,3 +36,48 @@ val partestSuite = Build.partestSuite
 val scalaTestSuite = Build.scalaTestSuite
 
 inThisBuild(Build.thisBuildSettings)
+
+val aggregatorIDs = Seq(
+  //"scalajs",
+  "ir",
+  "irJS",
+  "compiler",
+  "linkerInterface",
+  "linkerInterfaceJS",
+  //"linkerPrivateLibrary",
+  "linker",
+  "linkerJS",
+  "testAdapter",
+  //"sbtPlugin",
+  "javalanglib",
+  "javalib",
+  "scalalib",
+  "libraryAux",
+  "library",
+  "testInterface",
+  "testBridge",
+  "jUnitRuntime",
+  "jUnitTestOutputsJS",
+  "jUnitTestOutputsJVM",
+  "jUnitPlugin",
+  "jUnitAsyncJS",
+  "jUnitAsyncJVM",
+  "helloworld",
+  "reversi",
+  "testingExample",
+  "testSuite",
+  "testSuiteJVM",
+  "testSuiteEx",
+  "testSuiteLinker",
+  "partest",
+  "partestSuite",
+  "scalaTestSuite"
+)
+
+// Alias for developer usage
+addCommandAlias("scalastyle",  
+  "; scalajs/test:scalastyle; linkerPrivateLibrary/test:scalastyle; sbtPlugin/test:scalastyle" + 
+  ";" + aggregatorIDs.map(id => s"${id}2_11/test:scalastyle").mkString(";") +
+  ";" + aggregatorIDs.map(id => s"${id}2_12/test:scalastyle").mkString(";") +
+  ";" + aggregatorIDs.map(id => s"${id}2_13/test:scalastyle").mkString(";")
+)
```

Comment:
We already have machinery for this in `Build.scala`. Look for `allProjects`. Adding `scalastyle` should be a single line change.

Reply:
@gzm0 thanks for the pointer.   However, I'm struggling with the correct SBT configuration since `scalastyle` isn't a `TaskKey` (https://github.com/scalastyle/scalastyle-sbt-plugin/blob/master/src/main/scala/org/scalastyle/sbt/Plugin.scala#L51)  

```
import org.scalastyle.sbt.ScalastylePlugin.autoImport._
val scalaStyleTask = taskKey[Unit]("scalaStyle")
scalaStyleTask := scalastyle.toTask("").value,

...

        val keys = Seq[TaskKey[_]](
            clean, headerCreate in Compile, headerCreate in Test,
            headerCheck in Compile, headerCheck in Test, scalaStyleTask in Test
        )

```

Results iin  something like:

```
[error]
[error]   scalaTestSuite2_12 / Test / scalaStyleTask from Test / scalaStyleTask (Build.scala:587)
[error]
[error]   scalaTestSuite2_13 / Test / scalaStyleTask from Test / scalaStyleTask (Build.scala:587)
[error]
[error]   scalastyle from scalaStyleTask ((build.Build.root) Build.scala:559)
[error]      Did you mean Compile / scalastyle ?

```

Any ideas?

Reply:
I didn't realize `scalaStyle` is an `InputTask`. If you want to define a `Task` like this, you need to put the task definition into `commonSettings` so it gets applied to all projects (n.b. you need to do this in all configurations).

Reply:
@gzm0 I tried this, but couldn't get it working - spent several hours trying  to without luck. In the example code above, I have the `scalaStyleTask := scalastyle.toTask("").value,` in the command settings, or do you mean the actual val?

Reply:
Is there any reason not to include the  scalastyle in the normal compile (e.g. https://www.scala-sbt.org/1.x/docs/Howto-Dynamic-Task.html) I guess I was just trying to add it as a separate task to avoid making any "compile" changes, but this seems more straightforward

Reply:
Yes, there is a reason: because we don't want to spend time running the Scalastyle checker every time we compile ;)

Reply:
@sjrd that's fair, and also validates my intuition on not going down that path.  I *think* just figured it out, PR is updated, I'm doing some additional testing, is it worth updating the travis build to use this instead of the manual list?